models/esm2/tests/test_convert_reverse.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+from pathlib import Path
+
+import pytest
+import torch
+from torch import nn
+from transformers import AutoModelForMaskedLM
+
+
+def test_esm_model_has_all_te_layers():
+    """Test that the converted TE model doesn't contain vanilla PyTorch layers."""
+    from esm.convert import convert_esm_hf_to_te
+
+    model_hf = AutoModelForMaskedLM.from_pretrained("facebook/esm2_t6_8M_UR50D")
+    model_te = convert_esm_hf_to_te(model_hf)
+    vanilla_layers_found = []
+    for name, module in model_te.named_modules():
+        if isinstance(module, nn.Linear):
+            vanilla_layers_found.append(f"Linear layer found in {name}")
+        if isinstance(module, nn.LayerNorm):
+            vanilla_layers_found.append(f"LayerNorm layer found in {name}")
+    if vanilla_layers_found:
+        print("ERROR: Found vanilla PyTorch layers in converted TE model:")
+        for error in vanilla_layers_found:
+            print(f"WARNING: {error}")
+        assert not vanilla_layers_found, f"Found {len(vanilla_layers_found)} vanilla layers in converted model"
```

Comment:
this is a duplicate test, right? we have `test_esm_model_has_all_te_layers` in `test_modeling_esm_te`? do we need this in a test that exports from the TE model to HF?